code/datums/martial/krav_maga.dm

```diff
@@ -151,22 +155,27 @@
 /datum/martial_art/krav_maga/disarm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
 	if(check_streak(A,D))
 		return 1
-	var/obj/item/I = null
-	if(prob(60))
-		I = D.get_active_held_item()
-		if(I)
-			if(D.temporarilyRemoveItemFromInventory(I))
-				A.put_in_hands(I)
-		D.visible_message("<span class='danger'>[A] disarms [D]!</span>", \
-						"<span class='userdanger'>You're disarmed by [A]!</span>", "<span class='hear'>You hear aggressive shuffling!</span>", COMBAT_MESSAGE_RANGE, A)
-		to_chat(A, "<span class='danger'>You disarm [D]!</span>")
-		playsound(D, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
-	else
-		D.visible_message("<span class='danger'>[A] fails to disarm [D]!</span>", \
-						"<span class='userdanger'>You're nearly disarmed by [A]!</span>", "<span class='hear'>You hear a swoosh!</span>", COMBAT_MESSAGE_RANGE, A)
-		to_chat(A, "<span class='warning'>You fail to disarm [D]!</span>")
-		playsound(D, 'sound/weapons/punchmiss.ogg', 25, TRUE, -1)
-	log_combat(A, D, "disarmed (Krav Maga)", "[I ? " removing \the [I]" : ""]")
+	var/obj/item/bodypart/affecting = D.get_bodypart(ran_zone(A.zone_selected))
+	var/armor_block = D.run_armor_check(affecting, "melee")
+	if((D.mobility_flags & MOBILITY_STAND))
+		D.visible_message("<span class='danger'>[A] reprimands [D]!</span>", \
+					"<span class='userdanger'>You're slapped by [A]!</span>", "<span class='hear'>You hear a sickening sound of flesh hitting flesh!</span>", COMBAT_MESSAGE_RANGE, A)
+		to_chat(A, "<span class='danger'>You jab [D]!</span>")
+		A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+		playsound(D, 'sound/effects/hit_punch.ogg', 50, TRUE, -1)
+		D.apply_damage(rand(5,10), STAMINA, affecting, armor_block)
```

Comment:
you might want to make this disarm_act also apply shove slowdown (normal shoves slow people who(m) they hit down for a bit).

Reply:
You already have legsweep. Just legsweep. I would rather it have parity with your standard punch over trying to incorporate too much of the shove power back into what is literally a decent stamina punch. Given stamina damage already reduces your run speed, having it apply a slow in of itself would make it obscenely more powerful than even the harm stomps for sticking on people when I want that to be what legsweep enables.

The reason why I made it a nonlethal strike is because legsweep does what shove does for everyone else.